### PR TITLE
Separate satellite prefix from controller

### PR DESCRIPTION
**Pull request by riya-17**
PBENCH-464

This PR deals with removing satellite-prefix from controllers of the tarballs that are sent through  a  satellite server.

### CODE REVIEW

**Review by ndokos — DISMISSED** [review]
IIUC, this is needed in order to allow the PUT to the server that the put-shim does to succeed. Otherwise, the prefix with the double colon ends up in the `controller` header and is then checked by the server as a hostname,  but the check fails.

**Comment by ndokos** [comment]
This will need rebasing after #3181 is merged.

**Review by ndokos — DISMISSED** [review]

File: server/bin/gold/test-5.2.txt
Comment (webbnh) [inline server/bin/gold/test-5.2.txt:363]: Are we sad to lose these reporting distinctions?  Are we sad enough to try to do anything about it?

File: server/bin/gold/test-5.2.txt
Comment (webbnh) [inline server/bin/gold/test-5.2.txt:378]: We are no longer getting the "satellite percentages" here...should we correct the summary header texts?  (Do we care?)

**Review by webbnh — DISMISSED** [review]
I'm fine with this if no one cares about the reporting stuff.

File: server/bin/gold/test-5.2.txt
Comment (riya-17) [reply server/bin/gold/test-5.2.txt:378]: I don't know after removing the prefix from the controller name how we will be able to distinguish between a dataset at audit level. I am not sure if we want to pursue this.

File: server/bin/gold/test-5.2.txt
Comment (riya-17) [reply server/bin/gold/test-5.2.txt:363]: I don't know after removing the prefix from the controller name how we will be able to distinguish between a dataset at audit level. I am not sure if we want to pursue this.

Comment (dbutenhof) [reply server/bin/gold/test-5.2.txt:363]: This *will* be possible, after #3194 lands, and `pbench-results-move` and then put-shim leverage it, if we make the audit mechanism read the metadata. Maybe we just worry about it then. The whole "audit" process as current implemented is really archaic anyway, and needs to be completely rebuilt to consider SQL, Elasticsearch, Audit log, and so forth. I don't think this is worth worrying about now.

**Review by dbutenhof — DISMISSED** [review]

**Review by portante — CHANGES_REQUESTED** [review]
I am not sure about all this.

File: server/bin/pbench-sync-satellite.sh
Comment (portante) [inline server/bin/pbench-sync-satellite.sh:149]: Let's not change the legacy behavior.  We just need the put shim to strip it off.

**Review by portante — CHANGES_REQUESTED** [review]

File: server/bin/gold/test-5.2.txt
Comment (webbnh) [reply server/bin/gold/test-5.2.txt:378]: See [above](https://github.com/distributed-system-analysis/pbench/pull/3187#discussion_r1082851464).

File: server/bin/pbench-sync-satellite.sh
Comment (webbnh) [reply server/bin/pbench-sync-satellite.sh:149]: Let me know if, based on this change of direction, I should resolve the [conversation above](https://github.com/distributed-system-analysis/pbench/pull/3187#discussion_r1082851464).

Comment (riya-17) [reply server/bin/pbench-sync-satellite.sh]: Hey @webbnh the PR has changed, your reviews doesn't align with new approach so maybe we can resolve those.

Comment (ndokos) [reply server/bin/pbench-sync-satellite.sh]: I'm confused: `pbench-sync-satellite.sh` is *NOT* changing to put the tarballs in the reception area? It is still putting them in the archive? How does `put-shim` figure out where to get them?

Comment (portante) [reply server/bin/pbench-sync-satellite.sh]: > I'm confused: `pbench-sync-satellite.sh` is _NOT_ changing to put the tarballs in the reception area? It is still putting them in the archive? How does `put-shim` figure out where to get them?

That was implemented in PR #3181.

Comment (ndokos) [reply server/bin/pbench-sync-satellite.sh]: D'oh - thanks!

**Review by ndokos — APPROVED** [review]

File: server/lib/pbench/process_tb.py
Comment (ndokos) [inline server/lib/pbench/process_tb.py:101]: `satellite_prefix` is unused. I take it the plan is to add it to the metadata eventually?

**Review by ndokos — CHANGES_REQUESTED** [review]

File: server/lib/pbench/process_tb.py
Comment (riya-17) [reply server/lib/pbench/process_tb.py:101]: yes, that is the plan.

**Review by ndokos — APPROVED** [review]

**Review by portante — CHANGES_REQUESTED** [review]
I don't think we want to just remove the prefix from the controller but still send a controller string along with the API.  The API needs to change to drop the controller name entirely.

**Comment by dbutenhof** [comment]
> I don't think we want to just remove the prefix from the controller but still send a controller string along with the API. The API needs to change to drop the controller name entirely.

Sure; but I assume this was deliberately decoupled from that change to keep it small and focused. Yeah, we could block this until after #3193 lands, and expand it to also change `pbench-results-push` and `results.py` to remove the controller; I'd rather keep that separate and avoid the unnecessary dependency. (Also, this is 0.69 server code, while *that* is Agent `main`, so it's on different branches and can't really be a single PR anyway.)

**Review by dbutenhof — APPROVED** [review]

**Comment by portante** [comment]
> Sure; but I assume this was deliberately decoupled from that change to keep it small and focused. Yeah, we could block this until after #3193 lands, and expand it to also change `pbench-results-push` and `results.py` to remove the controller; I'd rather keep that separate and avoid the unnecessary dependency. (Also, this is 0.69 server code, while _that_ is Agent `main`, so it's on different branches and can't really be a single PR anyway.)

This change doesn't do anything for us, does it?  Can this 0.69 code base send data to new pbench server because of this change?  Yes, using controller header.  But we are getting rid of that entirely.  So why are we making any changes to the b0.69 branch that we know are not going to be what we want long term?  I just don't see the point of making such changes to the b0.69 branch.

**Comment by dbutenhof** [comment]
> > Sure; but I assume this was deliberately decoupled from that change to keep it small and focused. Yeah, we could block this until after #3193 lands, and expand it to also change `pbench-results-push` and `results.py` to remove the controller; I'd rather keep that separate and avoid the unnecessary dependency. (Also, this is 0.69 server code, while _that_ is Agent `main`, so it's on different branches and can't really be a single PR anyway.)
> 
> This change doesn't do anything for us, does it? Can this 0.69 code base send data to new pbench server because of this change? Yes, using controller header. But we are getting rid of that entirely. So why are we making any changes to the b0.69 branch that we know are not going to be what we want long term? I just don't see the point of making such changes to the b0.69 branch.

It allows the current put-shim server to successfully send satellite-relay data to the current 1.0 server, without dependency on the whole chain of removing the `controller` header from `PUT` and from the Agent CLI and class chain. It was a point solution to allow put-shim and satellite testing to proceed.

Sure, we can defer all that and wait for the real solution, but this small mitigation seemed worthwhile at the time.

**Review by webbnh — APPROVED** [review]
I'm OK with this if everyone else is.

**Review by portante — CHANGES_REQUESTED** [review]
How will the satellite prefix land as metadata for the tar ball on the PUT API call?

**Comment by riya-17** [comment]
> How will the satellite prefix land as metadata for the tar ball on the PUT API call?

This PR handles that, https://github.com/distributed-system-analysis/pbench/pull/3247

**Pull request closed without merging**
